pkg/storage/cache.go

```diff
@@ -53,42 +51,43 @@ var _ InMemoryCache[any] = (*InMemoryLRUCache[any])(nil)
 func NewInMemoryLRUCache[T any](opts ...InMemoryLRUCacheOpt[T]) *InMemoryLRUCache[T] {
 	t := &InMemoryLRUCache[T]{
 		maxElements: defaultMaxCacheSize,
-		closeOnce:   &sync.Once{},
+		stopOnce:    &sync.Once{},
 	}
 
 	for _, opt := range opts {
 		opt(t)
 	}
 
-	t.ccache = ccache.New(ccache.Configure[T]().MaxSize(t.maxElements))
+	var err error
+	t.client, err = theine.NewBuilder[string, T](t.maxElements).Build()
+	if err != nil {
+		panic(err)
```

Comment:
open to suggestions on how to improve this bit.

- rename to `MustNewInMemoryLRUCache` to signal that it can panic?
- change func signature to also return an error.
- keep as is.

Reply:
would advise 2 if the callers can handle the error, otherwise 1

Reply:
Return the error.

Reply:
[`6601cac` (#2117)](https://github.com/openfga/openfga/pull/2117/commits/6601caca349a5711bdfd67d38de7e6b228996243)